tower/src/discover/mod.rs

```diff
@@ -12,7 +12,8 @@
 //! # Examples
 //!
 //! ```rust
-//! use futures_util::{future::poll_fn, pin_mut};
+//! use std::future::poll_fn;
+//! use futures_util::pin_mut;
```

Comment:
The standard library also has the `pin!` macro that can replace this. That may also make sense to update in this PR.

(This isn't the only file using `pin_mut`.)

Reply:
Not possible without bumping MSRV from 1.64 to 1.68.

Reply:
This time, I have addressed this without updating the MSRV. I will update the MSRV and also address `pin_mut` if it is more appropriate.